server/api/src/rsGenQuery.cpp

```diff
@@ -650,7 +668,7 @@ rsGenQuery( rsComm_t *rsComm, genQueryInp_t *genQueryInp,
         // the zone name must be skipped when getZoneHintForGenQuery() derives it.
         if (getValByKey(&genQueryInp->condInput, ZONE_KW)) {
             if (!is_zone_name_valid(zone_hint_str)) {
-                rodsLog(LOG_ERROR, "%s:%d :: Unknown zone name [%s].", __func__, __LINE__, zone_hint_str.data());
+                log_api::error("[{}:{}] - Unknown zone name [{}].", __func__, __LINE__, zone_hint_str.c_str());
```

Comment:
You can remove the `.c_str()` since `zone_hint_str` is a `std::string`.

Reply:
Indeed. Should I port this back to main as well?

Reply:
I think so primarily for cherry-picking reasons. Keeping them in sync makes dealing with multiple branches easier.

Reply:
These changes are already merged in main. This *is* the cherry-pick. That being said, I don't mind porting this change over to main.

Reply:
Oh, if this is identical to what's in main, then leave it as is. No need to change anything.